tracer/src/Datadog.Trace/Util/EnumerableExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// <copyright file="EnumerableExtensions.cs" company="Datadog">
```

Comment:
Consider moving this to `Datadog.Trace/ExtensionMethods` (and changing its namespace) to keep them consolidated.

Reply:
(I know, we have `Datadog.Trace/Util/TypeExtensions.cs`, and two separate `StringExtensions.cs` files, but we can tryyyyy)

Reply:
You tricked me, I assumed `Datadog.Trace/ExtensionMethods` already existed, but you want a _new_ folder? 🤨 Meh 😝

Reply:
Hmm? No, not a new folder. https://github.com/DataDog/dd-trace-dotnet/tree/master/tracer/src/Datadog.Trace/ExtensionMethods

Reply:
oh, um, err.... 🙈

Reply:
Moved it in https://github.com/DataDog/dd-trace-dotnet/pull/5713/commits/6d71f7d8740e4b4ae582720e4eafbc89d23fecdc!